src/client_features.rs

```diff
@@ -395,4 +449,36 @@ mod tests {
         let first_hash_from_second_read = second_read.xx3_hash().unwrap();
         assert_eq!(first_hash, first_hash_from_second_read);
     }
+
+    #[test]
+    fn merging_two_client_features_takes_both_feature_sets() {
+        let client_features_one = ClientFeatures {
```

Comment:
Should we consider the version of client features? I'm thinking in a bit of an edge case:
```
client features(version: 1, features: [a, b])
client features(version: 2, features: [a])
// should be equal to
client features(version: 2, features: [a])
```
This would be the case of removing a feature

Reply:
I don't believe this is likely to happen. The version is something we bump when we do a breaking change to the features response. In practise we don't actually use this property in any of the SDKs and it doesn't do anything useful for us in Edge

Reply:
Perfect, I was thinking of other type of versioning (closer to the db)